test/unit/connection_string.test.ts

```diff
@@ -761,4 +764,40 @@ describe('Connection String', function () {
     expect(() => new MongoClient('mango://localhost:23')).to.throw(/Invalid scheme/i);
     expect(() => new MongoClient('mango+srv://localhost:23')).to.throw(/Invalid scheme/i);
   });
+
+  describe('when deprecated options are used', () => {
+    it('useNewUrlParser emits a warning', async () => {
```

Comment:
Should we only be emitting a single warning per-option?  The current implementation will emit a warning every time the option is specified.

Related question: do we have a practice of when we use `emitWarning` versus `emitWarningOnce`?

Reply:
What do you mean by every time it is specified?

I think emitWarning makes sense here because, in the case of creating multiple MongoClients, you can use Node's `--trace-warning` argument to get a stack trace pointing to each usage.